src/Uno.UI/UI/Xaml/DependencyProperty.cs

```diff
@@ -28,16 +29,16 @@ public sealed partial class DependencyProperty
 	{
 		private readonly static DependencyPropertyRegistry _registry = new DependencyPropertyRegistry();
 
-		private readonly static TypeToPropertiesDictionary _getPropertiesForType = new TypeToPropertiesDictionary();
-		private readonly static NameToPropertyDictionary _getPropertyCache = new NameToPropertyDictionary();
+		private readonly static Dictionary<Type, DependencyProperty[]> _getPropertiesForType = new(FastTypeComparer.Default);
+		private readonly static Dictionary<PropertyCacheEntry, DependencyProperty> _getPropertyCache = new();
 
 		/// <summary>
 		/// A static <see cref="PropertyCacheEntry"/> used for lookups and avoid creating new instances. This assumes that uses are non-reentrant.
 		/// </summary>
 		private readonly static PropertyCacheEntry _searchPropertyCacheEntry = new();
 
 
-		private readonly static FrameworkPropertiesForTypeDictionary _getFrameworkPropertiesForType = new FrameworkPropertiesForTypeDictionary();
+		private readonly static Dictionary<Uno.CachedTuple<Type, Windows.UI.Xaml.FrameworkPropertyMetadataOptions>, DependencyProperty[]> _getFrameworkPropertiesForType = new();
 
 		private readonly PropertyMetadata _ownerTypeMetadata; // For perf consideration, we keep direct ref the metadata for the owner type
 		private readonly PropertyMetadataDictionary _metadata = new PropertyMetadataDictionary();
```

Comment:
This is quite interesting to optimize.

In most cases, `_metadata` is going to be single element only. It only starts to have multiple elements when `OverrideMetadata` is called, which is not the common case.

This field should potentially be lazy, and when null we only consider `_ownerTypeMetadata`.